internal/app/productinfo/api/handlers.go

```diff
@@ -0,0 +1,104 @@
+package api
+
+import (
+	"net/http"
+
+	"fmt"
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+)
+
+// getServices retrieves services supported by the gibven provider in the given region
+func (r *RouteHandler) getServices(c *gin.Context) {
+
+	pathParams := getPathParams(c)
+	log.Debug("Path params: [%s]", pathParams)
+
+	var err error
```

Comment:
This error will never have a value and will always be nil. The redeclared `err` variable on the following line will only be valid within the if scope.

Also, in Go the happy path usually ends at the end of the function and error cases are handled earlier. So in this case errors should be checked for and handled first, like this:

```go
infoer, err := r.prod.GetInfoer(pathParams.Provider)
if err != nil {
    c.JSON(http.StatusInternalServerError, fmt.Sprintf("error while retrieving services: %s", err.Error()))

    return
}

services, err := infoer.GetServices(pathParams.Region)
if err != nil {
    c.JSON(http.StatusInternalServerError, fmt.Sprintf("error while retrieving services: %s", err.Error()))
    
    return
}

c.JSON(http.StatusOK, services)
```

Code duplication is less of an issue here.